cms/static/js/views/video_transcripts.js

```diff
@@ -0,0 +1,270 @@
+                    'transcriptUploadFailed'
+                );
+            },
+
+            /*
+            Sorts object by value and returns a sorted array.
+            */
+            sortByValue: function(itemObject) {
+                var sortedArray = [];
+                _.each(itemObject, function(value, key) {
+                    // Push each JSON Object entry in array by [value, key]
+                    sortedArray.push([value, key]);
+                });
+                return sortedArray.sort();
+            },
+
+            /*
+            Returns transcript title.
+            */
+            getTranscriptClientTitle: function() {
+                var clientTitle = this.clientVideoID;
+                // Remove video file extension for transcript title.
+                _.each(this.videoSupportedFileFormats, function(videoFormat) {
+                    clientTitle = clientTitle.replace(videoFormat, '');
+                });
+                return clientTitle.substring(0, 20);
+            },
+
+            /*
+            Toggles Show/Hide transcript button and transcripts container.
+            */
+            toggleShowTranscripts: function() {
+                var $transcriptsWrapperEl = this.$el.find('.show-video-transcripts-wrapper');
+
+                // Toggle show transcript wrapper.
+                $transcriptsWrapperEl.toggleClass('hidden');
+
+                // Toggle button text.
+                HtmlUtils.setHtml(
+                    this.$el.find('.toggle-show-transcripts-button-text'),
+                    StringUtils.interpolate(
+                        gettext('{toggleShowTranscriptText} transcripts ({totalTranscripts})'),
+                        {
+                            toggleShowTranscriptText: $transcriptsWrapperEl.hasClass('hidden') ? gettext('Show') : gettext('Hide'), // eslint-disable-line max-len
+                            totalTranscripts: _.size(this.transcripts)
+                        }
+                    )
+                );
+
+                // Toggle icon class.
+                if ($transcriptsWrapperEl.hasClass('hidden')) {
+                    this.$el.find('.toggle-show-transcripts-icon').removeClass('fa-caret-down').addClass('fa-caret-right'); // eslint-disable-line max-len
+                } else {
+                    this.$el.find('.toggle-show-transcripts-icon').removeClass('fa-caret-right').addClass('fa-caret-down'); // eslint-disable-line max-len
+                }
+            },
+
+            validateTranscriptUpload: function(file) {
+                var errorMessage = '',
+                    fileName = file.name,
+                    fileType = fileName.substr(fileName.lastIndexOf('.') + 1);
+
+                if (fileType !== this.videoTranscriptSettings.trancript_download_file_format) {
+                    errorMessage = gettext(
+                        '{filename} is not in a supported file format. ' +
```

Comment:
One more user facing string

Reply:
This file type is not supported. Supported file types are .srt and .txt. (If .srt and .txt are incorrect, replace those with the supported file type extensions.)